### ISSUE TITLE
[Task]: Clean up logger handling in Python's boot.go script

### ISSUE THREAD

**Issue by jrmccluskey** [apache/beam#29819]
### What needs to happen?

The Python SDK's [boot.go script](https://github.com/apache/beam/blob/60f040812403b3381e44f7d493aa2e9976767044/sdks/python/container/boot.go#L381) currently wraps [Logging API access in a buffer](https://github.com/apache/beam/blob/60f040812403b3381e44f7d493aa2e9976767044/sdks/python/container/boot.go#L381) to get appropriately leveled logging output sent across the logging API; however, the current structure accepts a pointer to a Logger then wraps it as needed; this should be cleaned up to pass around a buffered logger directly. 

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

**Comment by Tejasker** [comment]
Hii @jrmccluskey, I have read the above issue which needs slight modification in terms of using the logger ie rather than wrapping it in the buffer logger we can directly pass the bufferLogger which will be created outside the function and utilised as one of the parameter for `func installSetupPackages`  as required &
 I am thinking of the modified function signature as like this one

`func installSetupPackages(ctx context.Context, bufLogger *tools.BufferedLogger, files []string, workDir string, requirementsFiles []string)` 
`error {
    bufLogger.Printf(ctx, "Installing setup packages ...")// rest of the code
}`

I am assuming it would work. if this approach seems good to you then I can raise a PR for this issue

**Comment by Anuragt1104** [comment]
.take-issue

**Comment by ENate** [comment]
Hi @jrmccluskey Is the issue still open? Thanks.

**Comment by jrmccluskey** [comment]
Yes this issue is still open

**Comment by ENate** [comment]
Since it is a 'good first issue', I can work on it. But I will have to familiarize myself with the repo. Is that okay?

**Comment by jrmccluskey** [comment]
Yeah of course! It's fortunately a pretty straightforward change to pass the buffered logger around instead of a normal logger and re-wrapping it consistently. If you have any questions let me know!

**Comment by willvo2004** [comment]
.take-issue

**Comment by mohamedawnallah** [comment]
Hi @jrmccluskey, Is this issue still of interest? I noticed that @willvo2004's pull request #33015 was closed due to inactivity. If it's still relevant, I’d like to take it on. 🙏

**Comment by akulabs8** [comment]
.take-issue

**Comment by vedansh-5** [comment]
Hey @jrmccluskey is the issue still open? If so I'd like to take it on.

**Comment by jrmccluskey** [comment]
I think this only really applies to [`installSetupPackages`](https://github.com/apache/beam/blob/5da70fdf545919eb657b3bcb3e7cacfd683ae9f4/sdks/python/container/boot.go#L385) now, which gets invoked in setup-only mode as well (a context where we don't have a logging service to output to.) Going to close this one since it's out of date and gets a lot of attention better focused on other good-first-issue problems
